### DISCUSSION TITLE
With React Spectrum, how do I correctly fill the page background?

### DISCUSSION THREAD

**Discussion by SyntaxColoring** [adobe/react-spectrum#7255] [Q&A] [answered]
I'm following the [getting started guide](https://react-spectrum.adobe.com/react-spectrum/getting-started.html), so my code looks like this:

```tsx
function App() {
  return (
    (
      <Provider theme={defaultTheme}>
        <Button
          variant="accent"
          onPress={() => alert('Hey there!')}
        >
          Hello, React Spectrum!
        </Button>
      </Provider>
    )
  );
```

Out of the box, this creates an ugly partially-filled background:

<img width="567" alt="Screenshot 2024-10-24 at 7 45 14 PM" src="https://github.com/user-attachments/assets/e8d0bfe9-cb6d-4171-9597-792a040d1a98">

I can sort of fix that by setting `height: 100%` on `<html>`, `<body>`, the React root `<div>`, and the React Spectrum `<Provider>`. But then that doesn't cover overscroll:

https://github.com/user-attachments/assets/00c27781-9a1e-464d-829d-43d0457fcb7a

**Comment by mmorris8** [comment]
Setting the body background color to the same color as the app div should work.

**Reply by SyntaxColoring** [reply]
Are you suggesting doing something like `getComputedStyle()` on the div and then copying that to the `<body>` element?

Edit: This does not seem possible as far as I can tell. React gives us no reasonable way to monitor the div for changes to `getComputedStyle()`.

**Comment by snowystinger** [comment, marked as answer]
Are you trying to use React Spectrum to build a new Adobe app?
If this is a non-Adobe/Adobe-themed app, then you can do it any way you like using RAC instead https://react-spectrum.adobe.com/react-aria/index.html as it's bring your own styles.

Otherwise you can do something like https://codesandbox.io/p/sandbox/zk5sfz

I know it's a little clunky, it'll be easier in S2, but I don't know if you can use that yet.

**Reply by SyntaxColoring** [reply]
> Are you trying to use React Spectrum to build a new Adobe app?

Non-Adobe, but happy to accept Adobe styles and opinions for the sake of prototyping.

> If this is a non-Adobe/Adobe-themed app, then you can do it any way you like using RAC instead https://react-spectrum.adobe.com/react-aria/index.html as it's bring your own styles.

Totally. I'll get there eventually. :)

> Otherwise you can do something like https://codesandbox.io/p/sandbox/zk5sfz

On iOS, that technique has the unfortunate side effect of disabling the pull-to-refresh gesture. (It seems like it triggers when the root element scrolls.) May or may not be acceptable, depending on the app.

**Reply by snowystinger** [reply]
> On iOS, that technique has the unfortunate side effect of disabling the pull-to-refresh gesture. (It seems like it triggers when the root element scrolls.) May or may not be acceptable, depending on the app.

I hadn't noticed this, thanks for pointing it out. If this is just for prototyping and you're planning to move to RAC eventually with custom styling, it may be easier to start with https://react-spectrum.adobe.com/react-aria/getting-started.html#starter-kits as the API's between React Spectrum and RAC are not completely 1:1.

They'll be more similar in Spectrum 2 because we're building on top of RAC for that, but it's still in pre-release.

**Reply by cheema-corellian** [reply]
@snowystinger Is there any update on estimated release date/month for Spectrum 2?

I think I have seen an official estimate of "sometime in 2024".  I am wondering if this is still the case.

**Reply by snowystinger** [reply]
I don't have an exact date right now, but it's in prerelease, so you are welcome to try it out. Some APIs may be subject to change, but the code itself is pretty stable.
